src/rangefinder.jl

```diff
@@ -39,4 +39,27 @@ function rangefinder!(Y, A, Ω; q, Z, orthogonalize)
     end
 
     return nothing
-end
\ No newline at end of file
+end
+
+function chebyshev_rangefinder(A::AbstractMatrix{T}, ν::Float64, r::Int; q::Int=1, Ω=nothing, orthogonalize=true) where {T <: Number}
+    m, n = size(A)
+
+    isnothing(Ω) && (Ω = GaussianTestMatrix(m, r))
+
+    Y0 = qr(Ω.Ω).Q
+    Y1 = (2 / ν) * A * A' * Y0 - Y0
+    Y = hcat(Y0, Y1)
+
+    Yi_2 = Y0
+    Yi_1 = Y1
+    for i in 2:q
+        Yi = (4 / ν) * A * A' * Yi_1 - 2 * Yi_2 - Yi_2
+        Y = hcat(Y, Yi)
```

Comment:
Not sure if repeated use of `hcat` is bad practice in Julia (as opposed to e.g. allocating all the required space in a big matrix right at the start).

Reply:
Where possible, I would allocate at the beginning of the call since Y changes size and then overwrite columns.

Also use non-allocating matrix operations where possible (see https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#Low-level-matrix-operations). 

For example of doing the multiply, see https://github.com/tjdiamandis/RandomizedPreconditioners.jl/blob/85ed344725ac9fa6ace1812954ae3b08d0dd3f27/src/rangefinder.jl#L31